metricbeat/schema/schema.go

```diff
@@ -71,19 +71,13 @@ func (o Object) HasKey(key string) bool {
 
 // ApplyTo adds the fields extracted from data, converted using the schema, to the
 // event map.
-func (s Schema) ApplyTo(event common.MapStr, data map[string]interface{}) (common.MapStr, *Errors) {
```

Comment:
I would strongly prefer to use `error` in public interfaces. Using an specialized error type forces a strong coupling with the caller and makes the API more brittle. Callers that care about the precise error details can use a type switch or a type assertion to look for specific errors and extract details. Or for really common checks, the package can expose helper fucntions to inspect the error (e.g. `os.IsNotExists`, `os.IsPermission`).

This how most of the stdlib works and the methods that have specialized error types document what they return for callers that need the details (e.g. `os.Open` says "If there is an error, it will be of type *PathError."). 

This is a great [blog post](https://dave.cheney.net/2016/04/27/dont-just-check-errors-handle-them-gracefully) about the different error techniques.